CMakeLists.txt

```diff
@@ -69,7 +69,23 @@ include(Macros)
 include(CMakeDependentOption)
 include(CheckCCompilerFlag)
 
-project(CURL C)
+file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/include/curl/curlver.h" _curl_version_h_contents REGEX "#define LIBCURL_VERSION( |_NUM|_MAJOR|_MINOR|_PATCH )")
+string(REGEX MATCH "#define LIBCURL_VERSION \"[^\"]*" LIBCURL_VERSION ${_curl_version_h_contents})
+string(REGEX REPLACE "[^\"]+\"" "" LIBCURL_VERSION ${LIBCURL_VERSION})
+string(REGEX MATCH "#define LIBCURL_VERSION_NUM 0x[0-9a-fA-F]+" LIBCURL_VERSION_NUM ${_curl_version_h_contents})
+string(REGEX REPLACE "[^0]+0x" "" LIBCURL_VERSION_NUM ${LIBCURL_VERSION_NUM})
+string(REGEX MATCH "#define LIBCURL_VERSION_MAJOR ([0-9]+)" LIBCURL_MAJOR_VERSION ${_curl_version_h_contents})
+string(REGEX REPLACE "[^\"]+ " "" LIBCURL_MAJOR_VERSION ${LIBCURL_MAJOR_VERSION})
+string(REGEX MATCH "#define LIBCURL_VERSION_MINOR ([0-9]+)" LIBCURL_MINOR_VERSION ${_curl_version_h_contents})
+string(REGEX REPLACE "[^\"]+ " "" LIBCURL_MINOR_VERSION ${LIBCURL_MINOR_VERSION})
+string(REGEX MATCH "#define LIBCURL_VERSION_PATCH ([0-9]+)" LIBCURL_PATCH_VERSION ${_curl_version_h_contents})
+string(REGEX REPLACE "[^\"]+ " "" LIBCURL_PATCH_VERSION ${LIBCURL_PATCH_VERSION})
```

Comment:
What about replacing the above with this single line:
```cmake
string(REGEX REPLACE "([0-9]+\.[0-9]+\.[0-9]+).+" "\\1" CURL_VERSION_SEM "${CURL_VERSION}")
```
And then use `CURL_VERSION_SEM` as the bare/semantic version number.

It makes code shorter/faster and avoids breaking syntax highlighting 
for the rest of the file in certain editors (e.g. mcedit).

Reply:
OK. I originally think the major/minor/patch in `curlver.h` should be used.

Reply:
done